ci/unit_tests/dialog_xml2json/main_test.py

```diff
@@ -51,3 +53,38 @@ def test_mainValidActions(self):
 
         with open(expectedJsonPath, 'r') as expectedJsonFile, open(outputJsonPath, 'r') as outputJsonFile:
             assert json.load(expectedJsonFile) == json.load(outputJsonFile)
+
+
+    def test_mainMissingImport(self):
+        """Tests if the script fails with file with missing imported dialog file."""
+        inputXmlPath = os.path.abspath(os.path.join(self.dataBasePath, 'inputMissingImport.xml'))
+        inputXmlDir = os.path.dirname(os.path.abspath(inputXmlPath))
+        importedXmlPath = os.path.join(inputXmlDir, "nonexistentImport.xml")
+        self.t_exitCodeAndLogMessage(1, "Imported dialog file " + importedXmlPath + " not found.",
+                                    [['--common_dialog_main', inputXmlPath]])
+
+
+    def test_missingRoot(self):
+        """Tests if the script fails with file with missing root dialog file."""
+
+        self.t_exitCodeAndLogMessage(1, "Root dialog file nonexistentInputFile.xml not found.",
+                                    [['--common_dialog_main', 'nonexistentInputFile.xml']])
+
+
+    def test_validation(self):
+        """Tests if the script runs successfully with valid dialogs and fails with invalid dialog files."""
+
+        schemaPath = os.path.join(self.dataBasePath, "dialog_schema.xml")
+        validInputsFolder = os.path.join(self.dataBasePath, "validInputsAccordingToSchema")
+        invalidInputsFolder = os.path.join(self.dataBasePath, "invalidInputsAccordingToSchema")
+
+        for f in os.listdir(validInputsFolder):
```

Comment:
I'm still not sure if this is better than call each file separately.. You can then customize log messages.. And avoid weird filenames (goto_missingTarget.xml  when target present etc.)

Reply:
You are right, will fix this.